docs/Toy/toy/dialects/toy.py

```diff
@@ -87,6 +102,19 @@ def get_data(self) -> list[float]:
         return [float(el.value.data) for el in self.value.data.data]
 
 
+class InferAddOpShapeTrait(ToyShapeInferenceTrait):
+    @classmethod
+    def infer_shape(cls, op: Operation) -> None:
+        assert isinstance(op, AddOp)
+        assert isinstance(op.lhs.typ, TensorType)
```

Comment:
@math-fehr Do you reckon I should just bail out if the inputs aren't shaped correctly?

Reply:
What do you mean by that?

Reply:
I mean shape inference is a no-op if the inputs are not `TensorType`

Reply:
Just changed it not to crash, might want to not inline sometimes, things should still run